application/app/Http/Controllers/Admin/AdminUserController.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use App\Models\AdminUser;
+use Illuminate\Validation\Rule;
+use Illuminate\Support\Facades\Hash;
+
+class AdminUserController extends Controller
+{
+    public function index(Request $request)
+    {
+        //認可ポリシー追加　ログインユーザーがオーナーか判定
+        $this->authorize('index', AdminUser::class);
+
+        $adminUsers = AdminUser::query();
+
+        //リクエスト名前　判定　取得
+        if ($request->name !== null) {
+            //AdminUserスコープ追加
+            $adminUsers->partialName($request->name);
+        }
+
+        //リクエストメール　判定　取得
+        if ($request->email !== null) {
+            //AdminUserスコープ追加
+            $adminUsers->forwardEmail($request->email);
+        }
+
+        //リクエスト権限　判定　取得
+        if ($request->authority !== null) {
+            if ($request->authority === 'owner') {
+                $adminUsers->where('is_owner', true);
+            } else if ($request->authority === 'general') {
+                $adminUsers->where('is_owner', false);
+            }
+        }
+
+        //リクエスト　並び替え方向
+        switch ($request->sort_direction) {
+            case 'asc':
+                $direction = 'asc';
+                break;
+            case 'desc':
+                $direction = 'desc';
+                break;
+            default:
+                $direction = 'asc';
+        }
+
+        //リクエスト並び替え　判定
+        switch($request->sort_column) {
+            case 'id':
+                $adminUsers->orderBy('id', $direction);
+                break;
+            case 'name':
+                $adminUsers->orderBy('name', $direction);
+                break;
+            case 'email':
+                $adminUsers->orderBy('email', $direction);
+                break;
+            default:
+                $adminUsers->orderBy('id', 'asc');
+        }
+
+        //リクエスト表示項目数　判定
+        if ($request->page_unit === null) {
+            $adminUsers = $adminUsers->paginate(10);
+        } else {
+            $adminUsers = $adminUsers->paginate($request->page_unit);
+        }
+
+        return view('admin.admin_users.index', [
```

Comment:
```suggestion
        return view('admin.admin_users.index', compact('adminUser'));
```

https://www.php.net/manual/ja/function.compact.php

Reply:
なるほど